src/app/Library/CrudPanel/Traits/ColumnsProtectedMethods.php

```diff
@@ -146,15 +146,21 @@ protected function makeSureColumnHasEntity($column)
             // if the first part of the string exists as method,
             // it is a relationship
             if (method_exists($this->model, $possibleMethodName)) {
```

Comment:
@pxpm couldn't we add just this validation to the condition?
```suggestion
            if (method_exists($this->model, $possibleMethodName) && $this->makeSureMethodDoesNotRequireParameters($this->model, $possibleMethodName)) {
```

I'm having an hard time understanding the logic behind this conditions 😅

From what I understand by reading the code if the method requires params, entity = false, so isset = true, so, entity = false. 
Otherwise, it becomes column name even if the method require parameters ...

Isn't this wrong? 

By following the suggested code above, entity would only be name when method exists and takes no args.
Maybe I'm not seeing something, let me know ✌

Reply:
This applies for the other situations below.

Reply:
Hello @promatik 

Thanks for the review. I don't have the same understanding of you, I think. 

1 - This code only runs when `entity` is not set by developer, if it is this code is not run at all. 
2- From my understanding, if method **requires** parameters `entity => false`, so `isset($entity)` return `true` and it will continue to be false, otherwise, if the method does not require parameters it is a possible relation so we don't set entity and when checking if `isset()` would return false and asume the previous value we had, either `name` or `possible name`.

I've refactored the function name, moved it to the `CrudPanel` as we can also use that method for the same check in fields. 
Also refactored the if statement into a ternary condition. 

Once again, thanks for the review!

Pedro

Reply:
Love the new function name! `modelMethodHasParameters()` is much better than `makeSureMethodDoesNotRequireParameters()` 👏👏👏

It looks ok to me. It looks cleaner to me like it is now:
```
if (condition_1)
  variable = condition_2 ? value_1 : value_2;
  return column;
endif;
```

Otherwise, in order to do both conditions in one go, it would look something like:
```
if (condition_1 && condition_2)
   variable = value_1;
   return column;
else
   variable = value_2;
   return column;
endif;
```

Meh... I guess that's ok too. But I don't have a preference either way. If it works, I'll merge it this way 😀

Reply:
All good @promatik ? Should I merge this?

Reply:
@pxpm got it, it makes total sense now, thank you for the explanation ✌

@tabacitu, I read it once again, this is good to go 🙌